implot-wgpu-examples/src/main.rs

```diff
@@ -0,0 +1,235 @@
+                        );
+
+                        // Show individual examples in collapsed headers
+                        if CollapsingHeader::new(im_str!("Basic lineplot")).build(&ui) {
+                            ui::show_basic_plot(&ui, &plot_ui);
+                        }
+                        if CollapsingHeader::new(im_str!("Configurable lineplot")).build(&ui) {
+                            ui::show_configurable_plot(&ui, &plot_ui);
+                        }
+                        if CollapsingHeader::new(im_str!("Querying a plot")).build(&ui) {
+                            ui::show_query_features_plot(&ui, &plot_ui);
+                        }
+                        if CollapsingHeader::new(im_str!("Styling a plot")).build(&ui) {
+                            ui::show_style_plot(&ui, &plot_ui);
+                        }
+                        if CollapsingHeader::new(im_str!("Colormap selection")).build(&ui) {
+                            ui::show_colormaps_plot(&ui, &plot_ui);
+                        }
+                    });
+                }
+
+                if showing_demo {
+                    implot::show_demo_window(&mut showing_demo);
+                }
+
+                let mut encoder: wgpu::CommandEncoder =
+                    device.create_command_encoder(&wgpu::CommandEncoderDescriptor { label: None });
+
+                if last_cursor != Some(ui.mouse_cursor()) {
+                    last_cursor = Some(ui.mouse_cursor());
+                    platform.prepare_render(&ui, &window);
+                }
+
+                let mut rpass = encoder.begin_render_pass(&wgpu::RenderPassDescriptor {
+                    color_attachments: &[wgpu::RenderPassColorAttachmentDescriptor {
+                        attachment: &frame.output.view,
+                        resolve_target: None,
+                        ops: wgpu::Operations {
+                            load: wgpu::LoadOp::Clear(wgpu::Color {
+                                r: 0.1,
+                                g: 0.2,
+                                b: 0.3,
+                                a: 1.0,
+                            }),
+                            store: true,
+                        },
+                    }],
+                    depth_stencil_attachment: None,
+                });
+
+                renderer
+                    .render(ui.render(), &queue, &device, &mut rpass)
+                    .expect("Rendering failed");
+
+                drop(rpass); // renders to screen on drop, will probaly be changed in wgpu 0.7 or later
+
+                queue.submit(Some(encoder.finish()));
+            }
+            _ => {}
+        }
+
+        platform.handle_event(imgui.io_mut(), &window, &event);
+    });
+}
+
+fn main() {
+    let event_loop = EventLoop::new();
+    let window = winit::window::Window::new(&event_loop).unwrap();
+    // Temporarily avoid srgb formats for the swapchain on the web
```

Comment:
Are you building this to WASM, or is this just a generic comment from example code?

Reply:
I replaced the sample code with the one from https://github.com/Yatekii/imgui-wgpu-rs/blob/master/examples/hello_world.rs with some minor changes which IMO make it more succinct.